### DISCUSSION TITLE
Can Kerberos use lldap as a backend?

### DISCUSSION THREAD

**Discussion by dblsaiko** [lldap/lldap#754] [General]
Hi,

Is it possible to use lldap to store the KDC database (see https://web.mit.edu/kerberos/krb5-latest/doc/admin/conf_ldap.html)? I want to set up central auth (again) but this time trying to avoid the crusty OpenLDAP...

Thanks!

**Comment by nitnelave** [comment]
Hi!

I just had a quick look at the guide you pointed to, and I think it might
work: figuring out the details initially might take some work, but after
that the setup should be easier than open LDAP.

The main problem I see is loading the ldif file: we don't support that.
Depending on what's required, it will be possible or not. Altering the
LLDAP schema to add the attributes they mention in the ldif is possible
through user-defined attributes. Currently, the feature works but doesn't
have a good UI. You can make it work by crafting GraphQL queries directly,
see the schema at the root, and the docs/scripting.md (sorry for the lack
of links, on my phone). Make sure the attributes are lowercase (I'm working
on removing this requirement).

If they require specific (new) object classes, or anything other than users
and groups, it won't be possible.

ACLs are very basic: only your user data, read all, read all + reset
passwords, or write all (admin).

If you want to give it a try, feel free to join discord for some help (or
ask in this discussion).

If you're looking for a more fully-featured LDAP server to take the
classical LDAP route, check out kanidm: it's also written in Rust, and the
author is a really nice guy who also helped with LLDAP. It's a bit overkill
for a homelab, but so is openLdap.

Cheers and good luck!

**Comment by dblsaiko** [comment]
Thank you so much! I'm actually looking into kanidm too, going to try both and see which one fits better. I'll post updates when I figure something out :D

**Reply by ivan-pinatti** [reply]
Hey @2xsaiko, were you able to test IF lldap works well with Kerberos? 😃

**Reply by dblsaiko** [reply]
Not yet, sorry. I had to dismantle part of my network including the KDC and haven't gotten around to setting it back up yet.
